pkg/ccl/backupccl/backup_job.go

```diff
@@ -1521,7 +1521,11 @@ func getTenantInfo(
 		// coming from the tenants table and we should only
 		// ever have valid tenant IDs returned to us.
 		prefix := keys.MakeTenantPrefix(roachpb.MustMakeTenantID(tenants[i].ID))
-		spans = append(spans, roachpb.Span{Key: prefix, EndKey: prefix.PrefixEnd()})
+		// We append MaxKey to the tenant's prefix rather than using PrefixEnd since
+		// we want an EndKey that remains within the tenant's prefix and does not
+		// equal the next tenant's start key, as that can cause out spans to get
+		// merged but they need to remain separate for backup prefix elision.
+		spans = append(spans, roachpb.Span{Key: prefix, EndKey: append(prefix, keys.MaxKey...)})
```

Comment:
We now have this in like 3 places. Perhaps we should move it to a named function somewhere that takes a tenant ID and returns this type of span.

Reply:
yeah, i plan to remove it OR call sites. Keeping this as a draft for now until i fix the interesting failures in CI.